### ISSUE TITLE
Failed to maximize the window to specific zone

### ISSUE THREAD

**Issue by JianuoHuang** [digitalcreations/MaxTo#407]
Maximize to zone feature works fine for a while. But after I deactivated and then activated the program, the feature failed to work. The window maximize button started to maximize to the whole screen instead of each zone. But（Win)+（direction keys）can move the window around in different zones. Restart app or reactive the app cannot solve the problem. Restart the PC does fix the problem.

**Comment by vegardlarsen** [comment]
Thank you for reporting this; it may very well be that MaxTo failed to re-activate the hook it uses to intercept these events. 

- Can you check the logs for what happens when you re-activate?
- Does this happen every time you deactivate and reactivate?

**Comment by JianuoHuang** [comment]
Hello, thanks for the reply. I just tested out and found that just simply deactivate and then re-activate will not cause this problem. However, if a maximize window action was executed between the deactivation and the re-activation the situation will happen.
thanks

**Comment by vegardlarsen** [comment]
Thank you for the update. The logs from around the time this happens would still be very useful. [Instructions can be found here](https://maxto.net/documentation/troubleshooting).

**Comment by JianuoHuang** [comment]
Thanks, actually there is more thing I want to address, please give me a few minutes.

**Comment by JianuoHuang** [comment]
Also, there are actually more things related to that. You probably already aware that the Win+Shift+S snipping tool is breaking while using Max. Meanwhile, I found when I am using an app called Drawboard PDF, Maxto will conflict with the copy function and cause collapse. (no matter using the copy button in the app or Ctrl+C). I am not sure if this also happens on some other apps but after deactivating the Maxto, everything working fine again. 
So get back to the Maximize to zone feature, just like what I mentioned above, but not only maximize a window in between re-activation will cause the problem, doing anything like Win+Shift+S snipping tool and copying test in the Drawboard app in between will cause the problem as well. But there are some minor differences, take Win+Shift+S snipping tool as an example, use it in between, and then reactivate Maxto will cause no effect, but one the toll was used again when Maxto is activating, the problem will happen.
Interestingly, once the maximize to zone feature problem happens, all the problem I mention in the paragraph one will be automatically “fixed”, which means the snipping tool and copying feature will working fine when the maximize to zone feature is failed. 
 
I will attach the log later.

**Comment by JianuoHuang** [comment]
The log is attached below, 
And btw, Win+Shift+S snipping tool in between sometimes will not cause the problem.

2019-05-08 16:39:25 [Core@2.0.1.499] [Error] Process Server exited without explanation
2019-05-08 16:39:25 [UserInterface@2.0.1.499] [Information] Deactivated "MaxTo.UI.ViewModels.NotificationIconViewModel". Closed: True
2019-05-08 16:40:03 [Core@2.0.1.499] [Information] Logger setup (online logging: True, process #10288)
2019-05-08 16:40:03 [Core@2.0.1.499] [Information] Found no specified configuration location file, using default settings location "C:\Users\huang\AppData\Roaming\MaxTo"
2019-05-08 16:40:04 [Core@2.0.1.499] [Information] Logger setup (online logging: True, process #10288)
2019-05-08 16:40:04 [Core@2.0.1.499] [Information] Read shims from "C:\Users\huang\AppData\Local\MaxTo\shims.json"
2019-05-08 16:40:04 [Core@2.0.1.499] [Information] Squirrel-installed app, so rewriting CLI batch file.
2019-05-08 16:40:07 [Core@2.0.1.499] [Information] No updates found
2019-05-08 16:40:08 [Server@2.0.1.499] [Information] Logger setup (online logging: True, process #13460)
2019-05-08 16:40:09 [Core@2.0.1.499] [Information] Server up and running, client connected.
2019-05-08 16:40:10 [CompanionX64@2.0.1.499] [Information] Logger setup (online logging: True, process #14116)
2019-05-08 16:40:10 [CompanionX64@2.0.1.499] [Information] Found no specified configuration location file, using default settings location "C:\Users\huang\AppData\Roaming\MaxTo"
2019-05-08 16:40:10 [Core@2.0.1.499] [Information] 4 processes started
2019-05-08 16:40:10 [CompanionX64@2.0.1.499] [Information] Read shims from "C:\Users\huang\AppData\Local\MaxTo\shims.json"
2019-05-08 16:40:10 [CompanionX86@2.0.1.499] [Information] Logger setup (online logging: True, process #14072)
2019-05-08 16:40:11 [CompanionX64@2.0.1.499] [Information] Enabled hook
2019-05-08 16:40:11 [CompanionX86@2.0.1.499] [Information] Found no specified configuration location file, using default settings location "C:\Users\huang\AppData\Roaming\MaxTo"
2019-05-08 16:40:12 [CompanionX86@2.0.1.499] [Information] Read shims from "C:\Users\huang\AppData\Local\MaxTo\shims.json"
2019-05-08 16:40:12 [CompanionX86@2.0.1.499] [Information] Enabled hook
2019-05-08 16:40:12 [UserInterface@2.0.1.499] [Information] Logger setup (online logging: True, process #14256)
2019-05-08 16:40:12 [UserInterface@2.0.1.499] [Information] Activated "MaxTo.UI.ViewModels.NotificationIconViewModel"
2019-05-08 16:40:29 [CompanionX64@2.0.1.499] [Information] Disabled hook
2019-05-08 16:40:29 [CompanionX86@2.0.1.499] [Information] Disabled hook
2019-05-08 16:40:42 [CompanionX86@2.0.1.499] [Information] Enabled hook
2019-05-08 16:40:42 [CompanionX64@2.0.1.499] [Information] Enabled hook
2019-05-08 16:41:00 [CompanionX64@2.0.1.499] [Information] Disabled hook
2019-05-08 16:41:00 [CompanionX86@2.0.1.499] [Information] Disabled hook
2019-05-08 16:41:17 [CompanionX86@2.0.1.499] [Information] Enabled hook
2019-05-08 16:41:17 [CompanionX64@2.0.1.499] [Information] Enabled hook
2019-05-08 16:41:43 [CompanionX64@2.0.1.499] [Information] Disabled hook
2019-05-08 16:41:43 [CompanionX86@2.0.1.499] [Information] Disabled hook
2019-05-08 16:41:51 [UserInterface@2.0.1.499] [Information] Found 2 screens.
2019-05-08 16:41:58 [CompanionX86@2.0.1.499] [Information] Enabled hook
2019-05-08 16:41:58 [CompanionX64@2.0.1.499] [Information] Enabled hook

**Comment by jonathantwong** [comment]
To add another data point, I see a similar (but also sporadic) issue where I'm unable to get certain windows (MS Word, Excel, Skype, etc) to maximize properly when holding down Shift, nor when use the Win+Arrows recipes. However, I am still able to use the Win+Numpad recipes to manipulate all windows as expected.

Here is my log of when this issue is occurring (but does not capture when it begins):
```
2019-05-09 14:53:04 [Core@2.0.1.499] [Information] Logger setup (online logging: True, process #18928)
2019-05-09 14:53:04 [Core@2.0.1.499] [Information] Found no specified configuration location file, using default settings location "%appdata%\Roaming\MaxTo"
2019-05-09 14:53:04 [Core@2.0.1.499] [Information] Logger setup (online logging: True, process #18928)
2019-05-09 14:53:04 [Core@2.0.1.499] [Information] Read shims from "%appdata%\Local\MaxTo\shims.json"
2019-05-09 14:53:04 [Core@2.0.1.499] [Information] Squirrel-installed app, so rewriting CLI batch file.
2019-05-09 14:53:05 [Core@2.0.1.499] [Information] No updates found
2019-05-09 14:53:05 [Core@2.0.1.499] [Information] Was not set to start elevated, but was started elevated. Let's run with scissors and accept what we are given. Continuing with elevation.
2019-05-09 14:53:06 [Server@2.0.1.499] [Information] Logger setup (online logging: True, process #17584)
2019-05-09 14:53:06 [Core@2.0.1.499] [Information] Server up and running, client connected.
2019-05-09 14:53:06 [Core@2.0.1.499] [Information] 4 processes started
2019-05-09 14:53:06 [CompanionX86@2.0.1.499] [Information] Logger setup (online logging: True, process #18124)
2019-05-09 14:53:06 [CompanionX64@2.0.1.499] [Information] Logger setup (online logging: True, process #17832)
2019-05-09 14:53:06 [CompanionX86@2.0.1.499] [Information] Found no specified configuration location file, using default settings location "%appdata%\Roaming\MaxTo"
2019-05-09 14:53:06 [CompanionX64@2.0.1.499] [Information] Found no specified configuration location file, using default settings location "%appdata%\Roaming\MaxTo"
2019-05-09 14:53:07 [CompanionX86@2.0.1.499] [Information] Read shims from "%appdata%\Local\MaxTo\shims.json"
2019-05-09 14:53:07 [CompanionX86@2.0.1.499] [Information] Enabled hook
2019-05-09 14:53:07 [UserInterface@2.0.1.499] [Information] Logger setup (online logging: True, process #18016)
2019-05-09 14:53:07 [CompanionX64@2.0.1.499] [Information] Read shims from "%appdata%\Local\MaxTo\shims.json"
2019-05-09 14:53:07 [CompanionX64@2.0.1.499] [Information] Enabled hook
2019-05-09 14:53:07 [UserInterface@2.0.1.499] [Information] Activated "MaxTo.UI.ViewModels.NotificationIconViewModel"
2019-05-09 14:53:17 [Core@2.0.1.499] [Information] Executing recipe '"Move window left"' with 1 ingredients
2019-05-09 14:53:17 [Core@2.0.1.499] [Information] Recipe '"Move window left"' executed successfully.
2019-05-09 14:53:17 [Core@2.0.1.499] [Information] Executing recipe '"Move window right"' with 1 ingredients
2019-05-09 14:53:17 [Core@2.0.1.499] [Information] Recipe '"Move window right"' executed successfully.
2019-05-09 14:53:18 [Core@2.0.1.499] [Information] Executing recipe '"Move window up"' with 1 ingredients
2019-05-09 14:53:18 [Core@2.0.1.499] [Information] Recipe '"Move window up"' executed successfully.
2019-05-09 14:53:22 [Core@2.0.1.499] [Information] Executing recipe '"Move window right"' with 1 ingredients
2019-05-09 14:53:22 [Core@2.0.1.499] [Information] Recipe '"Move window right"' executed successfully.
2019-05-09 14:53:23 [Core@2.0.1.499] [Information] Executing recipe '"Move window left"' with 1 ingredients
2019-05-09 14:53:23 [Core@2.0.1.499] [Information] Recipe '"Move window left"' executed successfully.
2019-05-09 14:53:25 [Core@2.0.1.499] [Information] Executing recipe '"Center window"' with 1 ingredients
2019-05-09 14:53:25 [Core@2.0.1.499] [Information] Recipe '"Center window"' executed successfully.
2019-05-09 14:53:29 [Core@2.0.1.499] [Information] Received exit signal, waiting for subprocesses to shut down
2019-05-09 14:53:29 [CompanionX64@2.0.1.499] [Information] Disabled hook because of exit
2019-05-09 14:53:29 [CompanionX86@2.0.1.499] [Information] Disabled hook because of exit
2019-05-09 14:53:29 [Server@2.0.1.499] [Information] Received exit message. Exiting in 250ms...
2019-05-09 14:53:29 [UserInterface@2.0.1.499] [Information] Deactivated "MaxTo.UI.ViewModels.NotificationIconViewModel". Closed: True
2019-05-09 14:53:29 [Server@2.0.1.499] [Warning] Lost connection to "ce169d65-5e88-4ed9-a64e-223029208d25"
System.IO.IOException: Pipe is broken.
   at System.IO.Pipes.PipeStream.WinIOError(Int32 errorCode)
   at System.IO.Pipes.PipeStream.BeginWriteCore(Byte[] buffer, Int32 offset, Int32 count, AsyncCallback callback, Object state)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.Stream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at MaxTo.Core.Communication.NamedPipeServer.<Send>d__26.MoveNext()
2019-05-09 14:53:29 [Server@2.0.1.499] [Warning] Lost connection to "e29a2159-0c9f-4927-b917-4b61465707b3"
System.IO.IOException: Pipe is broken.
   at System.IO.Pipes.PipeStream.WinIOError(Int32 errorCode)
   at System.IO.Pipes.PipeStream.BeginWriteCore(Byte[] buffer, Int32 offset, Int32 count, AsyncCallback callback, Object state)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.Stream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at MaxTo.Core.Communication.NamedPipeServer.<Send>d__26.MoveNext()
2019-05-09 14:53:29 [Server@2.0.1.499] [Warning] Lost connection to "de2ff48a-6740-49ee-9c31-b07054b5c6d3"
System.IO.IOException: Pipe is broken.
   at System.IO.Pipes.PipeStream.WinIOError(Int32 errorCode)
   at System.IO.Pipes.PipeStream.BeginWriteCore(Byte[] buffer, Int32 offset, Int32 count, AsyncCallback callback, Object state)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.Stream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at MaxTo.Core.Communication.NamedPipeServer.<Send>d__26.MoveNext()
2019-05-09 14:53:29 [Server@2.0.1.499] [Warning] Lost connection to "d3cb14a1-9167-4330-9c86-6d98d319ad7c"
System.IO.IOException: Pipe is broken.
   at System.IO.Pipes.PipeStream.WinIOError(Int32 errorCode)
   at System.IO.Pipes.PipeStream.BeginWriteCore(Byte[] buffer, Int32 offset, Int32 count, AsyncCallback callback, Object state)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.Stream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at MaxTo.Core.Communication.NamedPipeServer.<Send>d__26.MoveNext()
2019-05-09 14:53:29 [Server@2.0.1.499] [Warning] Lost connection to "1b8e6ed3-e839-421a-b407-2240fe65b326"
System.IO.IOException: Pipe is broken.
   at System.IO.Pipes.PipeStream.WinIOError(Int32 errorCode)
   at System.IO.Pipes.PipeStream.BeginWriteCore(Byte[] buffer, Int32 offset, Int32 count, AsyncCallback callback, Object state)
   at System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_0(Stream stream, ReadWriteParameters args, AsyncCallback callback, Object state)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncTrim[TInstance,TArgs](TInstance thisRef, TArgs args, Func`5 beginMethod, Func`3 endMethod)
   at System.IO.Stream.BeginEndWriteAsync(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.Stream.WriteAsync(Byte[] buffer, Int32 offset, Int32 count, CancellationToken cancellationToken)
   at MaxTo.Core.Communication.NamedPipeServer.<Send>d__26.MoveNext()
```

**Comment by vegardlarsen** [comment]
I was aware of the snipping tool issue (#386). I am able to reproduce the issue where de-activating MaxTo and re-activating it leads to applications not obeying regions:

1. Start Notepad
2. Ensure MaxTo works on that window
3. De-activate MaxTo
4. Maximize Notepad
5. Re-activate MaxTo
6. Notepad now appears to be unaffected by MaxTo

**Comment by gregchamblin** [comment]
I am also having issues with the Canary and Dev builds of Edge (Chromium) on 2 different PCs.

**Comment by JoPe72** [comment]
Also issues with Edge Chromium

**Comment by vegardlarsen** [comment]
@gregchamblin @JoPe72 The browser issues have been solved as part of #476.

**Comment by shawn-anderson** [comment]
Yes.  I just purchased the program 5 days ago and began having the same issue by perhaps the second day.  I sent you an email with details.

**Comment by vegardlarsen** [comment]
@JianuoHuang I believe this issue was solved as [part of 2.1.0-alpha.4](https://maxto.net/en/release/alpha), where we fixed a multi-threading issue in the code that could block the hook from re-activating.

Please can you get back to me if this issue is still present in that version?
